test/common/stats/stats_impl_test.cc

```diff
@@ -466,5 +472,148 @@ TEST(TagProducerTest, CheckConstructor) {
       "No regex specified for tag specifier and no default regex for name: 'test_extractor'");
 }
 
+class BlockRawStatDataAllocatorTest : public testing::Test {
+public:
+  void setup(int num_stats) {
+    Stats::RawStatData::configureForTestsOnly(options_);
+    MemoryHashSetOptions memory_hash_set_options_ = Server::memHashOptions(num_stats);
+    ptr_ = static_cast<uint8_t*>(
```

Comment:
We should probably move to the way Josh allocates the memory in his tests - see [here](https://github.com/envoyproxy/envoy/blob/master/test/common/common/shared_memory_hash_set_test.cc#L49).  Something like:

```
uint32_t num_bytes = MemoryHashSet<Stats::RawStatData>::numBytes(memory_hash_set_options_);
ptr_ = new uint8_t[num_bytes];
```

Side note: why are you multiplying the number of bytes times the number of stats?  That may be the source of your issues - you're squaring the capacity.

Reply:
Great catch! I've updated this.

Reply:
Just kidding, I updated this for HotRestartNopImpl. I'll update it here too.